### PR TITLE
Convert Changelog to Textreader

### PR DESCRIPTION
**Pull request by wolfreak99**
This is part of an ongoing project to take bits and peices of my previous pull requests (from back when I started doing stuff) and seperate them into their own PR, making them easier to review and test and implement.

While this PR may look a bit intense, the majority of changes is rewording instances of "Changelog" to "Textreader". The only added code is within the "Integrate filepath parameter".
I have also added the ability to open the changelog.txt in the console. Along with that, I have added the ability to open the contributors.md in the console for demonstrations sake, to show the potential use of this being a thing.

I'm hoping this is more easier to handle than it being merged with my previous About OpenRCT2 PR.

Once this gets a general approval, I can remove the ability to open contributors.md in the changelog (unless it is desired), as that is more of a demonstration.

### CODE REVIEW

File: src/openrct2/windows/textreader.c [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+static void window_textreader_dispose_file();
+
+static char *_textreaderText = NULL;
+static size_t _textreaderTextSize = 0;
+static char **_textreaderLines = NULL;
+static sint32 _textreaderNumLines = 0;
+static sint32 _textreaderLongestLineWidth = 0;
+char _textreaderFilePath[MAX_PATH];
+
+rct_window *window_textreader_open(const char * file_path)
```
Comment (janisozaur) [inline src/openrct2/windows/textreader.c]: It should not be a responsibility of a window to open a file. The `textreader` should only receive a buffer and its size, while reading the file should be done elsewhere.

Keep in mind there's now https://github.com/OpenRCT2/OpenRCT2/pull/5263 going on.

Comment (wolfreak99) [inline src/openrct2/windows/textreader.c]: maybe name it textfile reader? this is pretty much just the changelog opened up.. another idea is i could do a window_textreader_open(const char * file_path) and a window_textreader_open_raw(const char * text, size_t size) if needbe

Comment (wolfreak99) [inline src/openrct2/windows/textreader.c]: @janisozaur I am looking at the source of this and notice that it also needs the number of lines and the longest line width.. should finding those two parts be textreaders responsibility? or should it be in the part that parses the text?

Comment (janisozaur) [inline src/openrct2/windows/textreader.c]: There should be no file operations in `textreader` window. It should only receive the buffer and its size. If it needs figuring out longest line size, it should do so itself.

Comment (wolfreak99) [inline src/openrct2/windows/textreader.c]: I do see one possibility we could do for now, we could move the file operations over to util.c....  But I'm not sure how to handle the buffers and memory-allocated variables.. Theres a few things i could do, i could: 
- have the buffer memory get allocated, then passed to the file reading operation (reallocating won't cause memory leaks if i remember correctly), and then have the window dispose the memory when closing.
- use a static variable in util.c as well, but that would open the possibility of the textreader windows text showing something else if something later on were to use the file reading operation at the same time.

I prefer the first option but i'm just unsure of how to lay this out..

**Comment by Gymnasiast** [comment]
We just discussed this (too late, I know). We think it's a good idea, but we'd want to do another approach, especially w/r/t the handling of markdown. For that reason, I'll mark this as information and close it.

**Comment by wolfreak99** [comment]
@Gymnasiast thank you for leaving feedback and updating me on the stuff as it gets talked about. while i've been working on my own stuff and taking care of personal stuff, if there is anything i could do to help get this or any of the other stuff i left behind in, i can help out. i'm burned on ideas lately but if you can describe what you'd like to see done to this i can try to add to it. i'm doing better with my life right now and i'm happier than what i was when i quit.

**Pull request closed without merging**